packages/core-commands/src/admin-navigation-commands.js

```diff
@@ -1,9 +1,95 @@
 /**
  * WordPress dependencies
  */
-import { useCommand } from '@wordpress/commands';
+import { useCommand, useCommandLoader } from '@wordpress/commands';
 import { __ } from '@wordpress/i18n';
 import { plus } from '@wordpress/icons';
+import { getPath } from '@wordpress/url';
+import { store as coreStore } from '@wordpress/core-data';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useMemo } from '@wordpress/element';
+import { store as noticesStore } from '@wordpress/notices';
+import { privateApis as routerPrivateApis } from '@wordpress/router';
+
+/**
+ * Internal dependencies
+ */
+import { unlock } from './lock-unlock';
+
+const { useHistory } = unlock( routerPrivateApis );
+
+function useAddNewPageCommand() {
```

Comment:
Apologies if this is either premature or not relevant, but: since this is specific to how things should behave in the site editor, would it be worth considering moving this hook into the site editor packages, instead of including it in `core-commands`? That way we wouldn't need to import the `notices` package in this package either.

Here's where the edit site package registers its commands: https://github.com/WordPress/gutenberg/blob/2d34e2aaa56e52c379b01bed2df2344ba01c9bb4/packages/edit-site/src/components/app/index.js#L32

So potentially one way to do it would be for the site editor to re-register (or de-register and then register) the "add new page" command using its own implementation instead of the core one.

What do you think?

Reply:
Good question. 

That was one of the questions I was going to ask: whether to use the `unregisterCommand` in the site editor.

Not needing to import a required library might not be the reason I'd lean on, but rather given that the command is very site-editor specific.

Reply:
Oh, forgot to mention. The reason I left it was because there are site editor specific commands in `core-commands` already:

https://github.com/WordPress/gutenberg/blob/90a058f0457caf03e40a4b327bb18f0eb5aa0a5c/packages/core-commands/src/site-editor-navigation-commands.js

Reply:
> Oh, forgot to mention. The reason I left it was because there are site editor specific commands in core-commands already:

Ignore me. These are universal commands 😄

Reply:
Hrm, interesting! Might be worth getting a second opinion here — I'm a little torn. I like keeping things together, but I could imagine other parts of WP wouldn't need the extra logic for the site editor stuff, so I wonder why it's been included in the `core-commands` package 🤔

That said, I don't feel too strongly about it either way, and the logic would only affect the site editor, so I imagine this could be moved around after-the-fact if need be, too.

Reply:
I think I'm in your boat. I made an [assumption](https://github.com/WordPress/gutenberg/pull/65476#discussion_r1766193665) and jumped in. So maybe I'll split, then move back if need be 😄

Reply:
Sounds good 😊